Documentation/special-services/pod-mutator.md

```diff
@@ -0,0 +1,76 @@
+# PodMutator Service
+To provide pod mutation capabilities for finer-grained control over rkt containerization concepts and debugging needs, 
+this proposal introduces a new optional stage1 app/service with special semantics that is lightweight.
+
+When added to a pod manifest, the PodMutator service, which is just an app with special semantics, 
+will expose app-level operations to the outside by communicating with the systemd inside the pod.
+
+## Architecture
+
+PodMutator, in essence, is rkt application container manipulation logic on top systemd/systemd-nspawn calls.
+Meaning that starting an App is translated into pod systemd calls to start that app's services.
+
+![pod-mutator](pod-mutator.png)
+
+- **Note:** Normal apps inside a pod will not have access to PodMutator.
+- **Note:** By using a PodMutator, we'll be able to maintain an empty Pod (but not empty since it has the PodMutator app).
+
+## Network
+
+PodMutator will expose this interface through a special network namespace that will 
+only be used by special services such as itself (use default-restricted namespace?).
+And will be exposed using an RPC protocol
```

Comment:
Alternatively, instead of an additional special network interface or network namespace, PodMutator could listen on a file socket and the rktnetes code could connect to the file socket directly.